common/src/View/MapDocument.cpp

```diff
@@ -2039,65 +2039,53 @@ namespace TrenchBroom {
                         group.transform(transformation);
-                        transformResult[i] = {groupNode, Model::NodeContents{std::move(group)}};
+                        return std::make_pair(groupNode, Model::NodeContents{std::move(group)});
                     },
-                    [&](Model::EntityNode* entityNode) {
+                    [&](Model::EntityNode* entityNode) -> TransformResult {
                         auto entity = entityNode->entity();
                         entity.transform(transformation);
-                        transformResult[i] = {entityNode, Model::NodeContents{std::move(entity)}};
+                        return std::make_pair(entityNode, Model::NodeContents{std::move(entity)});
                     },
-                    [&](Model::BrushNode* brushNode) {
+                    [&](Model::BrushNode* brushNode) -> TransformResult {
                         const bool lockTextures = lockTexturesPref
                             || (Model::findContainingLinkedGroup(*brushNode) != nullptr);
 
                         auto brush = brushNode->brush();
-                        brush.transform(m_worldBounds, transformation, lockTextures)
-                            .and_then([&](){
-                                transformResult[i] = {brushNode, Model::NodeContents{std::move(brush)}};
-                            }).handle_errors([&](const Model::BrushError e) {
-                                std::stringstream ss;
-                                ss << "Could not transform brush: " << e;
-                                
-                                auto lock = std::unique_lock<std::mutex>{errorMutex};
-                                transformErrors.push_back(ss.str());
-                                transformFailed = true;
+                        return brush.transform(m_worldBounds, transformation, lockTextures)
+                            .and_then([&]() -> TransformResult {
+                                return std::make_pair(brushNode, Model::NodeContents{std::move(brush)});
                             });
                     },
-                    [&](Model::PatchNode* patchNode) {
+                    [&](Model::PatchNode* patchNode) -> TransformResult {
                         auto patch = patchNode->patch();
                         patch.transform(transformation);
-                        transformResult[i] = {patchNode, Model::NodeContents{std::move(patch)}};
+                        return std::make_pair(patchNode, Model::NodeContents{std::move(patch)});
                     }
                 ));
             });
 
-            if (transformFailed) {
-                for (const std::string& message : transformErrors) {
-                    error() << message;
+            bool transformFailed = false;
+            const auto nodesToUpdate = kdl::collect_values(transformResults, kdl::overload(
+                [&](const auto& e) {
+                    error() << e;
```

Comment:
We lost the "Could not transform brush: " message.

Reply:
This should probably be a ` [&](const Model::BrushError& e)` lambda that formats the message as before ( "Could not transform brush: ").

Reply:
Yes, that would be better, but then I just removed the `std::string` case in 526f8ff9a.